tcproxy/tcproxy.go

```diff
@@ -0,0 +1,73 @@
+// Package tcproxy provides a simple wrapper around the tcproxy executable.
+// After the generic worker is refactored into engines, plugins and a runtime,
+// tcproxy will be a plugin instead.
+package tcproxy
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os/exec"
+	"strconv"
+	"time"
+
+	tcclient "github.com/taskcluster/taskcluster-client-go"
+)
+
+// TaskclusterProxy provides access to a taskcluster-proxy process running on the OS.
+type TaskclusterProxy struct {
+	command  *exec.Cmd
+	HTTPPort uint16
+	Pid      int
+}
+
+// New starts a tcproxy OS process using the executable specified, and returns
+// a *TaskclusterProxy.
+func New(taskclusterProxyExecutable string, httpPort uint16, creds *tcclient.Credentials, taskID string) (*TaskclusterProxy, error) {
+	args := []string{
+		"--port", strconv.Itoa(int(httpPort)),
+		"--client-id", creds.ClientID,
+		"--access-token", creds.AccessToken,
+	}
+	if creds.Certificate != "" {
+		args = append(args, "--certificate", creds.Certificate)
+	}
+	if taskID != "" {
+		args = append(args, "--task-id", taskID)
+	}
+	l := &TaskclusterProxy{
+		command:  exec.Command(taskclusterProxyExecutable, args...),
+		HTTPPort: httpPort,
+	}
+	err := l.command.Start()
+	// Note - we're assuming here that if the process fails to launch we'll get
+	// an error. We should test this to be sure.
+	if err != nil {
+		return nil, err
+	}
+	l.Pid = l.command.Process.Pid
+	log.Printf("Started taskcluster proxy process (PID %v)", l.Pid)
+	// Just to be safe, let's make sure the port is actually active before returning.
+	err = waitForPortToBeActive(httpPort)
+	return l, err
+}
+
+func (l *TaskclusterProxy) Terminate() error {
+	defer func() {
+		log.Printf("Stopped taskcluster proxy process (PID %v)", l.Pid)
+	}()
+	return l.command.Process.Kill()
```

Comment:
should we set the command and pid back to their nil values to ensure further attempts to kill processes fail?

Reply:
That is a very good idea!